opslevel/resource_opslevel_check_repository_search.go

```diff
@@ -35,7 +37,11 @@ func resourceCheckRepositorySearchCreate(d *schema.ResourceData, client *opsleve
 	checkCreateInput := getCheckCreateInputFrom(d)
 	input := opslevel.NewCheckCreateInputTypeOf[opslevel.CheckRepositorySearchCreateInput](checkCreateInput)
 	input.FileExtensions = opslevel.RefOf(getStringArray(d, "file_extensions"))
-	input.FileContentsPredicate = *expandPredicate(d, "file_contents_predicate")
+	predicate := expandPredicate(d, "file_contents_predicate")
+	if predicate == nil {
+		return fmt.Errorf("unexpected: expanding predicate 'file_contents_predicate' returned nil")
```

Comment:
think about this error message for a second.  This is what the end user is going to read - they have no idea what `expanding` means and they have no idea why it might be `nil` especially if they've actually configured it.  We should write a better / more helpful error message.  What are the error cases with `expandPredicate` (if any) - can we catch each one inside `expandPredicate` and provide specific helpful information for that error case to the end user so they know exactly what they need to fix?

Reply:
@rocktavious you are correct, this error message makes no sense and would only frustrate end users. I introduced it because of a change I was making and reverted but this was leftover.

I am fixing this to instead return an error explaining that a valid `file_contents_predicate` was expected to be found in the configuration for this request, but there was none found. 

Changed error message to be:

```
		return fmt.Errorf("error reading 'file_contents_predicate' in config - a valid type and value are required")
```

which matches the schema that `expandPredicate()` tries to parse from a config.